solr/solrj/src/java/org/apache/solr/common/util/ExecutorUtil.java

```diff
@@ -313,6 +340,37 @@ public MDCAwareThreadPoolExecutor(
         RejectedExecutionHandler handler) {
       super(corePoolSize, maximumPoolSize, keepAliveTime, unit, workQueue, handler);
       this.enableSubmitterStackTrace = true;
+      this.beforeExecuteTask = NOOP;
+    }
+
+    public MDCAwareThreadPoolExecutor(
+        int corePoolSize,
+        int maximumPoolSize,
+        int keepAliveTime,
+        TimeUnit timeUnit,
+        BlockingQueue<Runnable> blockingQueue,
+        SolrNamedThreadFactory httpShardExecutor,
+        boolean enableSubmitterStackTrace) {
+      super(
+          corePoolSize, maximumPoolSize, keepAliveTime, timeUnit, blockingQueue, httpShardExecutor);
+      this.enableSubmitterStackTrace = enableSubmitterStackTrace;
+      this.beforeExecuteTask = NOOP;
+    }
+
+    public MDCAwareThreadPoolExecutor(
+        int i,
+        int maxValue,
+        long l,
+        TimeUnit timeUnit,
+        BlockingQueue<Runnable> es,
+        SolrNamedThreadFactory testExecutor,
+        boolean b) {
+      this(i, maxValue, l, timeUnit, es, testExecutor, b, NOOP);
+    }
+
+    @Override
+    protected void beforeExecute(Thread t, Runnable r) {
+      super.beforeExecute(t, this.beforeExecuteTask);
```

Comment:
super.beforeExecute doesn't do anything; it's a callback -- something the underlying ThreadPoolExecutor will call, not something we should call..

Reply:
I think you mean to actually invoke this.beforeExecuteTask.run()

Reply:
quite right, that's an ide insertion I forgot to change. Good catch.